### PR TITLE
chore(config): remove ability to disable insights

### LINKED ISSUE #2565: Remove ability to disable Insights
### Summary

Today we have `metrics.dataplane.enabled` and `metrics.zone.enabled` options that allows to disable Insights. Taking into account https://github.com/kumahq/kuma/pull/2526#discussion_r686795326, probably makes sense to get rid of these options.

### Additional Details & Logs

- Version
- Error logs
- Configuration
- Platform and Operating System
- Installation Method (Helm, kumactl, AWS CloudFormation, etc.)

### PR DESCRIPTION
**Pull request by lahabana**
### Summary

remove `kuma.metrics.dataplane.enabled` and `kuma.metrics.zone.enabled` as insights are required internally anyway.

### Issues resolved

Fix #2565

### Documentation

- config are not updated

### Testing

- [x] Unit tests
- ~~[ ] E2E tests~~
- ~~[ ] Manual testing on Universal~~
- ~~[ ] Manual testing on Kubernetes~~

### Backwards compatibility

- [x] Update [`UPGRADE.md`](UPGRADE.md) with any steps users will need to take
      when upgrading.
- ~~[x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.~~ BREAKING CHANGE: it's now impossible to disable insights and the related configuration was removed

### CODE REVIEW

File: pkg/config/app/kuma-cp/config.go [hunk 1]
```diff
@@ -53,7 +53,6 @@ func (m *Metrics) Validate() error {
 }
 
 type DataplaneMetrics struct {
-	Enabled           bool          `yaml:"enabled" envconfig:"kuma_metrics_dataplane_enabled"`
```
Comment (jpeach) [inline pkg/config/app/kuma-cp/config.go:56]: Will this field still be accepted (and ignored?)

File: UPGRADE.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ does not have any particular instructions.
 
 ## Upcoming release
 
+### Any type
+
+It is not possible to remove computing stats with: `kuma.metrics.dataplane.enabled` and `kuma.metrics.zone.enabled` (the configurations are removed).
+These stats are useful to Kuma internally and can't be turned off.
```
Comment (jpeach) [inline UPGRADE.md:14]: Maybe rephrase this:

```suggestion
The `kuma.metrics.dataplane.enabled` and `kuma.metrics.zone.enabled` configurations have been removed.
Kuma always generate the corresponding metrics.
```

**Review by jpeach — APPROVED** [review]

File: pkg/config/app/kuma-cp/config.go (hunk 1 above)
Comment (lahabana) [reply pkg/config/app/kuma-cp/config.go:56]: yes
